homeassistant/components/honeywell/__init__.py

```diff
@@ -89,31 +91,38 @@ async def _retry(self) -> bool:
         if self._client is None:
             return False
 
-        devices = [
+        matching_devices = [
             device
             for location in self._client.locations_by_id.values()
             for device in location.devices_by_id.values()
+            if device.name == current_device.name
         ]
 
-        if len(devices) == 0:
-            _LOGGER.error("Failed to find any devices")
+        if len(matching_devices) != 1:
+            _LOGGER.error(
+                "Failed to find device %s, it may have disconnected from TCC.",
+                current_device.name,
+            )
             return False
 
-        self.devices = devices
+        for device in self.devices:
+            if device.name == matching_devices[0].name:
+                device = matching_devices[0]
+
         return True
 
-    def _refresh_devices(self):
-        """Refresh each enabled device."""
+    async def _initialize_devices(self):
+        """Refresh each enabled device. Only used for adding devices when setting up the integration."""
         for device in self.devices:
-            device.refresh()
+            await self._hass.async_add_executor_job(device.refresh)
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    async def update(self) -> None:
+    async def update(self, current_device) -> None:
```

Comment:
This won't work. We have a throttle on this method. The next entity that calls this method won't be able to update its corresponding device.

Reply:
Reverted the change this comment was talking about.